### ISSUE TITLE
Ignore constant assignments from mocha-no-side-effect-code

### ISSUE THREAD

**Issue by astorije** [microsoft/tslint-microsoft-contrib#445]
Given the following file:

```ts
import { expect } from 'chai';
import { resolve } from 'path';

const FOOBAR = resolve(__dirname, '..', 'fixture', 'foobar.txt');

describe('foo', function() {
  it('should foo all the things', function() {
    expect(foo(FOOBAR)).to.equal(/* ... */);
  });
});
```

`mocha-no-side-effect-code` will complain:

```
Mocha test contains dangerous variable initialization. Move to before()/beforeEach(): FOOBAR = resolve(__dirna...
```

As this is a constant that gets set for the entire file, I was trying to ignore all constants, with something naive like this:

```yaml
mocha-no-side-effect-code:
  - true
  - ignore: 'const [A-Z_]+ =' # Ideally with '^\Sconst\S[A-Z_]+\S=' or something
```

But couldn't make it work.
By `console.log`ging `initializer.getText()` [here](https://github.com/Microsoft/tslint-microsoft-contrib/blob/master/src/mochaNoSideEffectCodeRule.ts#L174), I realized that only this was tested against the regex:

```
resolve(__dirname, '..', 'fixture', 'foobar.txt')
```

Therefore I have no way to test a constant (by UPPER_CASE convention) with a regex.

Would it be acceptable to automatically ignore all `const\S[A-Z_]+\S=` from this rule? If so, how could we do this?
I also noticed that `const FOOBAR = '42';` does _not_ fail, if I understand correctly [due to this](https://github.com/Microsoft/tslint-microsoft-contrib/blob/master/src/mochaNoSideEffectCodeRule.ts#L112-L115), so I'm wondering if `const FOOBAR = resolve(__dirname, '..', 'fixture', 'foobar.txt');` can be made more "constant-looking".

**Comment by JoshuaKGoldberg** [comment]
If I understand correctly, the rule is validating at the expression level (`3 + 4`) instead of at the variable level (`const foo = 3 + 4;`). If a config setting were added to stop walking when it reaches a `const` variable, would that solve your use case @astorije?

**Comment by astorije** [comment]
@JoshuaKGoldberg, I believe so, yeah.

**Comment by JoshuaKGoldberg** [comment]
☠️  *It's time!* ☠️ 

Per #876, this repository is no longer accepting feature pull requests. TSLint is being deprecated and we recommend you switch to https://typescript-eslint.io.

Thanks for open sourcing with us, everyone!
